h2/src/main/org/h2/result/LocalResult.java

```diff
@@ -401,8 +402,23 @@ private void applyOffsetAndLimit() {
                 rows.clear();
                 return;
             }
+            int to = offset + limit;
+            if (withTies && sort != null) {
+                int[] indexes = sort.getQueryColumnIndexes();
+                Value[] expected = rows.get(to - 1);
+                loop: while (to < rows.size()) {
+                    Value[] current = rows.get(to);
+                    for (int idx : indexes) {
+                        if (!expected[idx].equals(current[idx])) {
+                            break loop;
+                        }
```

Comment:
Arrays.equals

Reply:
It cannot be used here, only columns (expressions) from `ORDER BY` clause are compared.